asciidoctor-gradle-jvm/src/main/groovy/org/asciidoctor/gradle/jvm/AsciidoctorJBasePlugin.groovy

```diff
@@ -50,8 +58,50 @@ class AsciidoctorJBasePlugin implements Plugin<Project> {
                 @Override
                 void execute(Task task) {
                     ((DependencyReportTask) task).configurations = [asciidoctorj.configuration].toSet()
+
+                    task.doLast {
+                        project.logger.warn "${task.name} is deprecated. " +
+                            "Use pattern <asciidocTaskName>Dependencies instead to get specific task's dependencies"
+                    }
                 }
             })
         }
+
+        registerDependencyReportRules(project)
     }
+
+    private void registerDependencyReportRules(Project project) {
+        TaskContainer tasks = project.tasks
+        tasks.addRule(
+            '<asciidocTaskName>Dependencies: Report dependencies for AsciidoctorJ tasks'
+        ) { String targetTaskName ->
+            Matcher matcher = targetTaskName =~ DEPS_TASK_PATTERN
+            if (matcher.matches()) {
+                try {
+                    TaskProvider associate = TaskProvider.taskByTypeAndName(
+                        project,
+                        AbstractAsciidoctorTask,
+                        taskBaseName(matcher)
+                    )
+
+                    tasks.create(targetTaskName, DependencyReportTask, new Action<Task>() {
```

Comment:
In this case no, because the task is being created from a rule. The rule is invoked because the user explicitly requested the task and wants it executed.

Also, if a task is registered from a rule, the task will never get executed and Gradle will report it as non-existing. I do not know whether that should be considered a bug in Gradle, but it is the current behaviour